File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        maxx=0
        for i in range(len(height)):
            while l<r:
                w=r-l
                h=min(height[l],height[r])
                area=h*w
                maxx=max(area,maxx)
                if height[l]<height[r]:
                    l+=1
                else:
                    r-=1
        return maxx                

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max=0;
        int n=height.length;
        int left=0;
        int right=n-1;

        while(left<right){
            int h=Math.min(height[left],height[right]);
            int w=right-left;
            int area=h*w;
            max=Math.max(area,max);
            if(height[left]<height[right]){
                left+=1;
            }else{
                right-=1;
            }
        }
        return max;
        
    }
}